### DISCUSSION TITLE
NFS fails to mount at boot

### DISCUSSION THREAD

**Discussion by bnilsson11** [rern/rAudio#601] [Fixed Issues]
rAudio is one of the major joys of my life, but some quirks remain.

My NFS NAS server constantly fails to mount at rAudio (re)boot, but manual mount under System never fails.
Always red when opening System after boot, always green one second after the manual mount request.
A timeout issue? Order of stuff done during boot?

/BN

**Comment by cmh714** [comment]
does your NAS sleep or something like that? working here fine...maybe post what your fstab looks like

**Reply by bnilsson11** [reply]
My NAS does not sleep as far as I know.
This happens also when I am sure it is running.

**Reply by bnilsson11** [reply]
Here is my fstab:

PARTUUID=53f67d48-01  /boot  vfat  defaults,noatime  0  0
PARTUUID=53f67d48-02  /      ext4  defaults,noatime  0  0
192.168.1.115:/home/bnilsson/Musik  /mnt/MPD/NAS/HTPC-NFS  nfs  defaults,noauto,bg,soft,timeo=5  0  0

**Reply by cmh714** [reply]
Change noauto to auto

**Reply by cmh714** [reply]
Pretty sure thats your issue....

**Reply by ceejayemm** [reply]
Mine also includes 'noauto' - not added by me so must be part of the default setup ?.  Here is my 'fstab', the only difference is that I specifically add a 'rw' option

PARTUUID=58b3ea50-01  /boot  vfat  defaults,noatime  0  0
PARTUUID=58b3ea50-02  /      ext4  defaults,noatime  0  0
192.168.86.250:/export/Music/iTunes/Music  /mnt/MPD/NAS/NAS_Music  nfs  defaults,noauto,bg,soft,timeo=5,rw  0  0


Chris

**Reply by cmh714** [reply]
Yeah default is noauto but IIRC, a year or so back I had this issue, and changing to auto made it work....YMMV

**Reply by bnilsson11** [reply]
I tried this:

PARTUUID=53f67d48-01  /boot  vfat  defaults,noatime  0  0
PARTUUID=53f67d48-02  /      ext4  defaults,noatime  0  0
192.168.1.115:/home/bnilsson/Musik  /mnt/MPD/NAS/HTPC-NFS  nfs  defaults,auto,bg,soft,timeo=5  0  0

but it made no difference, still red after reboot.
I also tried timeo=30, no difference, so I put it back to 5.

Manual "mount -a" returns successful within one second.

/BN

**Reply by bnilsson11** [reply]
I found that "mount -a" is NOT mounting the NFS with "noauto". Interesting.

**Comment by ceejayemm** [comment]
My NFS music mount works fine after reboot. My NAS (OMV 5.x) sleeps overnight (22:00 to 07:00) and r-Audio 1 reconnects automatically each morning without any problems. 

Chris

**Reply by cmh714** [reply]
Same here....so not sure what his issue may be.

**Comment by ceejayemm** [comment]
@bnilsson11 - are you mounting NFS share via server name or IP number ?  I had problems with earlier versions of RA/rAudio-1 using a server name in the share mounts.  After much too-ing and fro-ing with Rern he persuaded me to mount via IP Number and I have had no problems since.

Chris

**Reply by cmh714** [reply]
I too use IP, always have

**Reply by bnilsson11** [reply]
I mount it using IP.

**Reply by bnilsson11** [reply]
The latest update (as of today 2021-11-14) did not improve things, I still have to re-mount manually after boot.
fstab:
192.168.1.115:/home/bnilsson/Musik  /mnt/MPD/NAS/HTPC-NFS  nfs  defaults,auto,bg,soft,timeo=5  0  0

Not a BIG issue, just an annoying inconvenience...

**Reply by bnilsson11** [reply]
To be complete, this is the response from "mount" command once the NFS NAS is mounted:
192.168.1.115:/home/bnilsson/Musik on /mnt/MPD/NAS/HTPC-NFS type nfs4 (rw,relatime,vers=4.2,rsize=1048576,wsize=1048576,namlen=255,soft,proto=tcp,timeo=5,retrans=2,sec=sys,clientaddr=192.168.1.187,local_lock=none,addr=192.168.1.115)

And this is the /etc/exports on the server side (lubuntu 20.04 LTS):
/home/bnilsson/Musik	rp4(ro,sync,no_subtree_check)

**Reply by bnilsson11** [reply]
It seems like the latest update fixed the issue. Or something changed here...
Anyway it is working now. Thanks.

/BN
